code/gui5.py

```diff
@@ -0,0 +1,204 @@
+# Running into errors with the do_clustering() funtion
+
+from clusterer import do_clustering
+from VectorModelBuilder import VectorModelBuilder
+import tkinter as tk
+from tkinter.filedialog import askopenfilename, askdirectory
+
+VMB_ARGS = (("Dataset", ""),("Count Method", "ngram"),("Weighting", "ppmi"),("Output Directory","../vector_data/"),("Output File Name","None"),("Value of N", "3"))
+CLUSTERER_ARGS = (("Stem of Files",""),("Output File","../vector_data/"),("V Scalar","1"),("Constrain Partition","False"),("Constrain PCS","False"))
+
+LABELS = ["  Run  ","Browse...","False","True"]
+FILE_TYPE = (("Text Files", "*.txt"),("All Files", "*.*"))
+WEIGHT_METHODS = ["ppmi", "probability", "conditional_probability", "pmi", "none"]
+
+# Two classes to help initiate the algorithm
+class VectorModelBuilder_tk(VectorModelBuilder):
+    def __init__(self,dataset,count_method,weighting,outdir,outfile,n):
+        super().__init__(dataset,count_method,weighting,outdir,outfile,n)
+
+    def run(self):
+        self.create_vector_model()
+        self.save_vector_model()
+
+class Clusterer_tk():
+    def __init__(self,input_file_stem, output_file, v_scalar, 
+                  constrain_partition,
+                  constrain_pcs):
+        self.input_file_stem = input_file_stem
+        self.output_file = output_file
+        self.v_scalar = v_scalar
+        self.constrain_partition = constrain_partition
+        self.constrain_pcs = constrain_pcs
+
+    def run(self):
+        self.v_scalar = int(self.v_scalar)
+        if (self.constrain_partition == "True"):
+            self.constrain_partition = True
+        else:
+            self.constrain_partition = False
+
+        if (self.constrain_pcs == "True"):
+            self.constrain_pcs = True
+        else:
+            self.constrain_pcs = False
+
+        do_clustering(self.input_file_stem,self.output_file,self.v_scalar,self.constrain_partition,self.constrain_pcs)
+
+# Setting up the root of the window
+window = tk.Tk()
+window.title("Distributional Learning")
+#window.columnconfigure(0,minsize=400,weight=1)
+#window.rowconfigure(0,weight=1)
+
+################### Vector Model Builder ################
+vmb_frame = tk.Frame(master=window)
+vmb_frame.grid(row=0,column=0,padx=10,pady=10,sticky="nswe")
+vector_model_builder = tk.Label(master=vmb_frame,text="Vector Model Builder")
+vector_model_builder.grid(row=0,column=0,sticky="w")
+# looping to populate labels
+j=1
+for i, text in enumerate(VMB_ARGS):
+    label = tk.Label(master=vmb_frame,text=VMB_ARGS[i][0])
+    label.grid(row=j,column=0,sticky="e")
+    j+=1
+
+# Argument 1 Dataset
+def open_arg1_browse():
+    """Open a file."""
+    filepath = askopenfilename(
+        filetypes=[("Text Files", "*.txt"),("All Files", "*.*")]
+    )
+    if not filepath:
+        return
+    arg_path_ent.delete("0", tk.END)
+    arg_path_ent.insert(tk.END, filepath)
```

Comment:
I think you're right that having the function definitions where they are improves readability, but I'd add a new line after them just to set them apart a bit more.